pytestqt/wait_signal.py

```diff
@@ -123,6 +122,9 @@ def _quit_loop_by_signal(self, *args):
         """
         quits the event loop and marks that we finished because of a signal.
         """
+        if self.callback:
+            if not self.callback(*args):
+                return  # parameter check did not pass
```

Comment:
Sounds good.
Another question is: should this also be implemented for `MultiSignalBlocker`? It would generally be interesting to get a string-dump (in human readable form) of all recorded signals and their parameters, including the points of time where things went wrong (e.g. in case of _strict_ order when the order was violated). Similarly, if `raising == False`, `blocker.recorded_signals` (or something similar) could contain a list of recorded signals together with their parameters.